### PR TITLE
Format documents with leading `---`

### PR DESCRIPTION
**Pull request by Gabriel439**
The context for this is: https://github.com/dhall-lang/dhall-haskell/issues/1864

Not only is this legal, but I believe it's actually more consistent with
how `---` was intended to be used (to separate each YAML document from
its preceding header).  The YAML specification also uses `---` in the
same way for its document-related examples.

### CODE REVIEW

**Comment by Gabriel439** [comment]
Note that I don't plan on merging this until https://github.com/clovyr/aeson-yaml/pull/6 is merged first

**Comment by sjakobi** [comment]
So it looks like `dhall-to-yaml` and `dhall-to-yaml-ng` will behave consistently with `--documents` then? If so, that's good! :)

Would you mind adding one or two tests to the test-suite to check, @Gabriel439?

Maybe include this one from the issue description:

> ```
> $ dhall-to-yaml --documents <<< '"foo"'
> ---
> foo
> ```

Right now it's not clear to me whether `dhall-to-yaml-ng` will actually behave like that (nor, frankly, whether it should).

**Comment by sjakobi** [comment]
> Right now it's not clear to me whether `dhall-to-yaml-ng` will actually behave like that (nor, frankly, whether it should).

To clarify that last bit:

The documentation for `--documents` currently refers only to the case of a Dhall list. It's not obvious to me that `--documents` should also affect the output when the input is not a list.

```
  --documents              If given a Dhall list, output a document for every
                           element
```

**Comment by Gabriel439** [comment]
I'm just waiting for `aeson-yaml` to be released to Hackage so that the test can depend on that instead of the GitHub package for parity in behavior

**Comment by sjakobi** [comment]
So this behaviour is unchanged, right?

```
$ echo '"foo"' | dhall-to-yaml --documents
foo
$ echo '"foo"' | dhall-to-yaml-ng --documents
foo
```

That makes sense to me.

(CC @jfroche)

**Comment by Gabriel439** [comment]
@sjakobi: The behavior is changed.  Both should emit a leading `---\n`

**Comment by sjakobi** [comment]
> @sjakobi: The behavior is changed. Both should emit a leading `---\n`

OK! I hadn't realized that. Maybe it would be good to clarify the CLI info for `--documents` then?

File: dhall-json/src/Dhall/JSON/Yaml.hs [hunk 1]
```diff
@@ -57,7 +57,7 @@ parseDocuments :: Parser Bool
 parseDocuments =
   Options.Applicative.switch
             (   Options.Applicative.long "documents"
-            <>  Options.Applicative.help "If given a Dhall list, output a document for every element"
+            <>  Options.Applicative.help "If given a Dhall list, output a document for every element.  Each document, including the first one, will be preceded by \"---\", even if there is only one document"
```
Comment (sjakobi) [inline dhall-json/src/Dhall/JSON/Yaml.hs:60]: This still sounds like the option will only modify the behaviour when the input is a list, although it actually outputs non-list inputs as documents too. Sorry for being unclear about this aspect before!

But maybe this is not an important distinction, and the clarification isn't actually needed.

**Comment by sjakobi** [comment]
Thanks @Gabriel439! With the new testcase, things are much clearer now. AppVeyor should succeed again once the new `aeson-yaml` version is pinned in the `stack.yaml`.

**Comment by Gabriel439** [comment]
Yeah, I'm using the AppVeyor build failure as a reminder that `aeson-yaml` needs to get on Hackage first before merging this

**Comment by patrickmn** [comment]
aeson-yaml 1.1.0.0 released with this (and only this) change; just going to follow the PVP more closely from now on

**Pull request merged**